### PR TITLE
feat(material-experimental/mdc-tabs): add default fitInkBarToContent option

### PR DESCRIPTION
**Pull request by andrewseguin**

### CODE REVIEW

File: src/material-experimental/mdc-tabs/tab-group.spec.ts [hunk 1]
```diff
@@ -718,6 +718,35 @@ describe('MatTabGroup with ink bar fit to content', () => {
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SimpleTabsTestApp);
+    fixture.detectChanges();
+  });
+
+  it('should set whether the ink bar fits to content', () => {
+    const tabElement = fixture.nativeElement.querySelector('.mdc-tab');
+    const contentElement = tabElement.querySelector('.mdc-tab__content');
+    const indicatorElement = tabElement.querySelector('.mdc-tab-indicator');
+    expect(indicatorElement.parentElement).toBe(contentElement);
```
Comment (crisbeto) [inline src/material-experimental/mdc-tabs/tab-group.spec.ts]: Might be a good idea to also add assertions that the values here are truthy.

File: src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.spec.ts [hunk 2]
```diff
@@ -355,6 +357,35 @@ describe('MatTabNavBar', () => {
   });
 });
 
+describe('MatTabNavBar with a default config', () => {
+      let fixture: ComponentFixture<TabLinkWithTabIndexBinding>;
```
Comment (crisbeto) [inline src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.spec.ts]: The indentation seems off here.

File: src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.ts [hunk 3]
```diff
@@ -81,8 +86,13 @@ export class MatTabNav extends _MatTabNavBase implements AfterContentInit {
                * @deprecated @breaking-change 9.0.0 `platform` parameter to become required.
                */
               @Optional() platform?: Platform,
-              @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode?: string) {
+              @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode?: string,
+              @Optional() @Inject(MAT_TABS_CONFIG) defaultConfig?: MatTabsConfig) {
     super(elementRef, dir, ngZone, changeDetectorRef, viewportRuler, platform, animationMode);
+    this.disablePagination = defaultConfig && defaultConfig.disablePagination != null ?
```
Comment (crisbeto) [inline src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.ts]: Isn't this one handled by the base class already?

File: src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.ts [hunk 4]
```diff
@@ -81,8 +86,13 @@ export class MatTabNav extends _MatTabNavBase implements AfterContentInit {
                * @deprecated @breaking-change 9.0.0 `platform` parameter to become required.
                */
               @Optional() platform?: Platform,
-              @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode?: string) {
+              @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode?: string,
+              @Optional() @Inject(MAT_TABS_CONFIG) defaultConfig?: MatTabsConfig) {
     super(elementRef, dir, ngZone, changeDetectorRef, viewportRuler, platform, animationMode);
```
Comment (crisbeto) [inline src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.ts]: Should we pass the `defaultConfig` into the super class?

File: src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.ts (hunk 3 above)
Comment (andrewseguin) [reply src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.ts]: Unfortunately not, it extends `MatPaginatedTabHeader` which doesn't handle the config. You may be thinking of the `_MatTabGroupBase` which does handle this

File: src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.ts (hunk 4 above)
Comment (andrewseguin) [reply src/material-experimental/mdc-tabs/tab-nav-bar/tab-nav-bar.ts]: This extends `MatPaginatedTabHeader` which doesn't take the config

File: src/material/tabs/tab-config.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  /**
+   * Whether the ink bar should fit its width to the size of the tab label content.
+   * This only applies to the MDC-based tabs.
+   */
+  fitInkBarToContent?: boolean;
+}
+
+/** Injection token that can be used to provide the default options the tabs module. */
+export const MAT_TABS_CONFIG = new InjectionToken<MatTabsConfig>('MAT_TABS_CONFIG');
```
Comment (jelbourn) [inline src/material/tabs/tab-config.ts]: I think this name is inconsistent with other components

File: src/material-experimental/mdc-tabs/tab-group.spec.ts (hunk 1 above)
Comment (andrewseguin) [reply src/material-experimental/mdc-tabs/tab-group.spec.ts]: Good point, I added an expectation that the indicator's parent element is truthy

File: src/material/tabs/tab-config.ts (hunk 5 above)
Comment (jelbourn) [reply src/material/tabs/tab-config.ts]: Based on in-person conversation, we'll file an issue for the inconsistent naming across components and address in the v10 range.

**Pull request merged**